controllers/metric_controller.go

```diff
@@ -138,11 +142,13 @@ func (r *MetricReconciler) evaluateMetrics(ctx context.Context, t *redskyv1beta1
 }
 
 func (r *MetricReconciler) collectMetrics(ctx context.Context, t *redskyv1beta1.Trial, probeTime *metav1.Time) (*ctrl.Result, error) {
-	// Index the metric definitions
+	// Fetch the experiment
 	exp := &redskyv1beta1.Experiment{}
 	if err := r.Get(ctx, t.ExperimentNamespacedName(), exp); err != nil {
 		return &ctrl.Result{}, err
 	}
+
+	// Index the metric definitions
```

Comment:
should the metrics collection be ( close to ) atomic or should we allow partial metrics to be collected as we go?

ex, if I have 5 metrics configured and the 5th metric is giving me an issue, does it make sense to recollect the previous 4 metrics each time or should we stash each metric in the status as we get the results?